nixos/modules/services/hardware/keyd.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = [ "cidkid" ];
+  };
+
+  options = {
+    services.keyd = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          enable keyd daemon for remapping keys
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keyd;
+        defaultText = literalExpression "pkgs.keyd";
+        description = ''
+          KeyD Package to use
+        '';
+      };
+
+      configuration = mkOption {
+        type = types.lines;
```

Comment:
Would be nice to allow a config file path as well.

Reply:
Not exactly filepath since `keyd` expects it's configs under `/etc/keyd` but I've made it possible to have multiple config files in the latest commit

Reply:
If it needs a directory containing files passed to it rather than a specific file, there still isn't anything wrong with setting this to `types.path`.